Automattic/wp-calypso
# (Site Editor)(wpcom-block-editor) Remove dotcom-specific Site Editor "manage all pages" click handler

## Proposed Changes

Proposal for fixing https://github.com/Automattic/wp-calypso/issues/87009. 

Related to: https://github.com/Automattic/wp-calypso/pull/81509, where this override was introduced.

The definitive view should be the new "DataVIew" view from core. I'm not 100% this is the right solution, though, as the  removed logic took into account settings that are available only in dotcom, so I'm not sure if we should deprecate it or fix it cc @mmtr 

If this is truly deprecated code, then we should probably also look into removing the `/wpcom/v2/admin-menu` endpoint and its code. Finally, if that's the case, then it seems we could also remove the (related?) https://github.com/Automattic/wp-calypso/blob/trunk/apps/wpcom-block-editor/src/wpcom/features/override-edit-site-back-button.js module? cc @jeyip 

## Testing Instructions

Considering you're running a dotcom WP simple instance with at least Gutenberg 17.6.2+ installed and active.

1. Reproduce the issue by following the steps in https://github.com/Automattic/wp-calypso/issues/87009. It might depend on the setting returned by `/wpcom/v2/admin-menu`, but I could reproduce it consistently;
2. Sandbox `widgets.wp.com`;
3. Checkout this branch in In your local `wp-calypso` copy, then go to `apps/wpcom-block-editor` and run `yarn dev --sync` to compile and sync it to your sandbox;
4. Follow the steps in the issue again and you shouldn't be able to reproduce - it should work as expected.
